### ISSUE TITLE
Ginkgo CI: Kubernetes tests need to wait for cilium endpoints to be regenerated

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#2075]
Parent: #1839 

Kubernetes tests need to wait until cilium endpoints are regenerated. For instance, in `test/k8sT/Services.go`:

```
It("Check Service with cross-node", func() {
		demoDSPath := fmt.Sprintf("%s/demo_ds.yaml", kubectl.ManifestsPath())
		kubectl.Apply(demoDSPath)
		defer kubectl.Delete(demoDSPath)

		svcIP, err := kubectl.Get(
			helpers.DefaultNamespace, "service testds-service").Filter("{.spec.clusterIP}")
		Expect(err).Should(BeNil())
		log.Debugf("svcIP: %s", svcIP.String())
		Expect(govalidator.IsIP(svcIP.String())).Should(BeTrue())

		url := fmt.Sprintf("http://%s/", svcIP)
		testHTTPRequest(url)
	})
```

The test performs no check to see whether endpoints are regenerated yet, which is necessary as the tests check their connectivity. Make sure all Kubernetes tests ensure that endpoints are regenerated.

**Comment by ianvernon** [comment]
See invocation of `wait_for_cilium_ep_gen` in `tests/k8s/*.sh`. This functionality should be matched in the Ginkgo tests.

**Comment by eloycoto** [comment]
Hi @ianvernon 

This is already fixed on f9482755819d74aa9fa65bc02e3d5d9f21581d01 commit. 


TBH, that is a bit weird, the wait for pods was always in there, see the original commit: 1c68e01a60dc97cad5dc36b550ebd62ced772408

Regards
